models/models_methods.go

```diff
@@ -655,7 +716,7 @@ func validateWorkflows(config *BitriseDataModel) ([]string, error) {
 				}
 
 				// TODO: Why is this assignment needed?
-				stepListItem[stepID] = step
+				stepListItem[stepID] = *step
```

Comment:
Here we reassigned the step with a pointer type, that's why we had to handle both pointer and value type steps here: